### PR TITLE
Add `View as:` toggle to unit overview page

### PR DESCRIPTION
**Pull request by lfryemason**
* Adds a new toggle to view page as student or teacher - this is moved from the teacher panel
* Moves `teacher resources` dropdown, `printing options` dropdown, `assign to sections` button and new view as toggle to a single row under the unit title
* Removes the `active section` selector
* Migrates some classes and components to modern best practices (scss vs styles, typescript etc)

### With teacher nav v2 disabled
**This will immediately effect all teachers on production, whether they are in the closed beta or not.**
Note - teacher panel is still enabled and a teacher will be able to see and use both interchangeably
image

### Teacher nav v2 enabled - unassigned version
image

### Teacher nav v2 enabled - teacher_dashboard version
image

## Links
https://codedotorg.atlassian.net/browse/TEACH-1446
https://codedotorg.atlassian.net/browse/TEACH-1345

## Testing story
Manually tested all buttons and changes on multiple courses on adhoc:

https://adhoc-lfm-n-view-toggle-studio.cdn-code.org/s/coursea-2024?section_id=1&viewAs=Instructor

### CODE REVIEW

File: apps/src/code-studio/components/progress/UnitOverviewHeader.jsx [hunk 1]
```diff
@@ -165,7 +156,6 @@ class UnitOverviewHeader extends Component {
             notice={i18n.wrongCourseVersionWarningNotice()}
             details={versionWarningDetails}
             dismissible={true}
-            width={SCRIPT_OVERVIEW_WIDTH}
```
Comment (lfryemason) [inline apps/src/code-studio/components/progress/UnitOverviewHeader.jsx:168]: Fixes announcements being wider than other page content and makes them width responsive.

File: apps/src/code-studio/components/progress/UnitOverviewHeader.jsx [hunk 2]
```diff
@@ -219,49 +202,28 @@ class UnitOverviewHeader extends Component {
   }
 }
 
-const styles = {
```
Comment (lfryemason) [inline apps/src/code-studio/components/progress/UnitOverviewHeader.jsx:222]: Moved to scss

File: apps/src/templates/teacherNavigation/selectedSectionLoader.ts [hunk 3]
```diff
@@ -66,8 +66,6 @@ export const setSelectedSectionData = (sectionData: any) => {
     getStore().dispatch(setShowSharingColumn(true));
   }
 
-  // TODO: update sections list with selected section data to ensure consistency.
```
Comment (lfryemason) [inline apps/src/templates/teacherNavigation/selectedSectionLoader.ts:69]: Already done

File: apps/test/unit/code-studio/components/progress/UnitOverviewHeaderTest.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import {shallow} from 'enzyme'; // eslint-disable-line no-restricted-imports
 import React from 'react';
 
-import {UnconnectedUnitOverviewHeader as UnitOverviewHeader} from '@cdo/apps/code-studio/components/progress/UnitOverviewHeader';
+import {UnconnectedUnitOverviewHeader} from '@cdo/apps/code-studio/components/progress/UnitOverviewHeader';
```
Comment (lfryemason) [inline apps/test/unit/code-studio/components/progress/UnitOverviewHeaderTest.js:4]: I really dislike this pattern of `UnconnectedUnitOverviewHeader as UnitOverviewHeader`. It obscures what is imported and requires the developer to look at the imports to understand that it is not using the connected version of the class.

File: apps/test/unit/code-studio/components/progress/UnitOverviewTopRowTest.jsx [hunk 5]
```diff
@@ -131,23 +129,6 @@ describe('UnitOverviewTopRow', () => {
     ).toBe(true);
   });
 
-  it('renders SectionAssigner for instructor', () => {
```
Comment (lfryemason) [inline apps/test/unit/code-studio/components/progress/UnitOverviewTopRowTest.jsx:134]: Removing tests that check for removed elements. Will add unit tests for UnitOverviewActionRow as a follow up

**Comment by MollyPeredo** [comment]
Looks good! What size is the "view page as" label? It looks slightly larger than the text on the buttons and it's drawing my eye to it. Maybe worth playing with reducing that by a point or two.

I'm also wondering if we should remove the printer icon from the Printing Options button. It's looking a little crowded. Not a top priority if it's more than like 5 minutes of work though.

File: apps/test/unit/code-studio/components/progress/UnitOverviewHeaderTest.js (hunk 4 above)
Comment (Nokondi) [reply apps/test/unit/code-studio/components/progress/UnitOverviewHeaderTest.js:4]: I agree

**Review by Nokondi — APPROVED** [review]
This looks good. You might add a section to the main PR comment about adding unit tests as follow up work. Do we already have a ticket for that?

**Comment by lfryemason** [comment]
> Looks good! What size is the "view page as" label? It looks slightly larger than the text on the buttons and it's drawing my eye to it. Maybe worth playing with reducing that by a point or two.


> I'm also wondering if we should remove the printer icon from the Printing Options button. It's looking a little crowded. Not a top priority if it's more than like 5 minutes of work though.
Done

**Review by Nokondi — APPROVED** [review]
Looks good!

**Pull request merged**